src/Adapter/Product/QueryHandler/GetEditableCombinationsListHandler.php

```diff
@@ -118,16 +133,36 @@ private function formatCombinationsForEditing(
                 }
             }
 
-            $combinationsForEditing[] = new CombinationForEditing(
+            $impactOnPrice = new DecimalNumber($combination['price']);
+            $combinationsForEditing[] = new EditableCombinationForListing(
                 $combinationId,
                 $this->buildCombinationName($combinationAttributesInformation),
-                $combinationAttributesInformation
+                $combinationAttributesInformation,
+                (bool) $combination['default_on'],
+                $impactOnPrice,
+                $this->calculateFinalPrice($product, $impactOnPrice),
+                (int) $combination['quantity']
```

Comment:
Careful with the quantity, it should be fetched via the associated `StockAvailable` quantity which is, supposedly, the true container of the product stock
From what I understood the quantity fields in product and combination is mostly a shortcut..

You can see how it is done for product here in `GetProductForEditingHandler:getProductStockInformation`:

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Product/QueryHandler/GetProductForEditingHandler.php#L296